src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -47,7 +51,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class IcebergInputFormat implements InputFormat {
+public class IcebergInputFormat implements InputFormat,  CombineHiveInputFormat.AvoidSplitCombination {
```

Comment:
I know you tried a bunch of things to get this working so just checking on which ones are still actually required - is `CombineHiveInputFormat.AvoidSplitCombination` one of them?

Reply:
yeah, this was what fixed the issue where our IF wasn't ever getting called: https://stackoverflow.com/questions/29133275/custom-inputformat-getsplits-never-called-in-hive

Reply:
OK, can you please add the above as a comment here for future reference?